wandb/apis/public.py

```diff
@@ -2237,6 +2237,45 @@ def entity(self):
 
     @property
     def state(self):
+        item = self.get_item()
+        if item:
+            return item["state"].lower()
+
+        raise ValueError(
+            f"Could not find QueuedRunItem associated with id: {self.id} on queue id {self.queue_id} at itemId: {self.id}"
+        )
+
+    @normalize_exceptions
+    def get_item(self):
+        query = gql(
+            """
+            query GetRunQueueItem($projectName: String!, $entityName: String!, $runQueue: String!, $itemId: ID!) {
+                project(name: $projectName, entityName: $entityName) {
+                    runQueue(name: $runQueue) {
+                        runQueueItem(id: $itemId) {
+                            id
+                            state
+                            associatedRunId
+                        }
+                    }
+                }
+            }
+        """
+        )
+        variable_values = {
+            "projectName": self.project,
+            "entityName": self._entity,
+            "runQueue": self.queue_id,
+            "itemId": self.id,
+        }
+        try:
+            res = self.client.execute(query, variable_values)  # exception w/ old server
+            if item := res["project"]["runQueue"].get("runQueueItem"):
```

Comment:
Does this run in all our python versions?

Reply:
Hmm, seeing as the tests just lit up like a firestation getting a call about a forest fire I'm going to assume no :(